src/shims/unix/fd.rs

```diff
@@ -189,6 +189,26 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
     }
 
+    fn ioctl(&mut self, args: &[OpTy<'tcx>]) -> InterpResult<'tcx, Scalar> {
+        let this = self.eval_context_mut();
+
+        let [fd_num, cmd] = check_min_arg_count("ioctl", args)?;
+        let fioclex = this.eval_libc_u64("FIOCLEX");
+
+        let fd_num = this.read_scalar(fd_num)?.to_i32()?;
+        let cmd = this.read_scalar(cmd)?.to_u64()?;
+
+        if cmd == fioclex {
+            if this.machine.fds.is_fd_num(fd_num) {
```

Comment:
```suggestion
            // Since we don't support `exec`, this is a NOP. However, we want to
            // return EBADF if the FD is invalid.
            if this.machine.fds.is_fd_num(fd_num) {
```